compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUTileTensor.cpp

```diff
@@ -92,24 +92,22 @@ static LogicalResult tileParallelDims(func::FuncOp funcOp,
   std::array<int64_t, 3> elementPerWorkgroup = {
       distributeToWarp ? workgroupSize[0] / kWarpSize : workgroupSize[0],
       workgroupSize[1], workgroupSize[2]};
-  SmallVector<Operation *> computeOps;
-  SmallVector<LoopTilingAndDistributionInfo> tiledLoops;
-  if (failed(getComputeOps(funcOp, computeOps, tiledLoops))) {
-    return funcOp.emitOpError("failed to get compute ops");
-  }
+  SmallVector<TilingInterface> computeOps;
+  funcOp.walk([&](TilingInterface op) { computeOps.push_back(op); });
```

Comment:
This is going to be walking the function through all nesting etc. How do you ensure that the ops picked up here are not scoped at the same level?

Reply:
each op is tiled individually (there is no tile and fuse at this level). This was already like that for the second level of LLVMGPU backend. 
I'm not sure I see the problem?

Reply:
Earlier this was using `getComputeOps`. It is very constrained in that it looks for ops within a block, and errors out if the dispatch deviates from that. This is intentional to avoid compilation go down paths that are uninteded. A walk is more free form. How do you guarantee that the ops collected here are all to be tiled at the same time. So its not the same.

Reply:
Right `walk` is what I want I think as I want to distribute all operations. I could look at the attribute but I don't think I need to as I always want to distribute all the ops across the group

Reply:
Asking differently, is there a case here where ops from different blocks need to be distributed. Most of the ops we use have a single block, so different blocks represent different levels of nesting. From that I am unsure distributing ops at different levels of nesting is valid... So I think flipping this to not distribute ops at different levels of nesting at the same time and erroring out inside avoids unintended lowering.

Reply:
We tile the reduction loop before doing the second level of tile and distribute (to handle shared memory promotion) so we do end up with linalg.matmul within a scf.for region while the fused ops would be the top level basic block. Both those needs to be distributed.

Reply:
Note that the reduction loop being outside causes other problems that I'm trying to solve (I shared a doc with you where I'm starting to discuss this) so ideally we would remove it at some point but for now I don't have an alternative solution to move the bufferization down.

Reply:
Yeah, that makes sense and gives me a better idea. Thanks for explaining!